### PR TITLE
refactor: own performance and TTY runtime metadata

### PR DESCRIPTION
**Pull request by nickna**
Performance and TTY primitives used unchecked flat declarations on the runtime container. Separate optional `EmittedPerformanceRuntime` and `EmittedTtyRuntime` components now own those handles, with independent feature availability, checked reads, and validation and freezing at completion.

- Narrow three emitter helpers to the owning components and the explicit numeric-conversion dependency. Update primitive consumers and remove the flat aliases.
- Preserve declaration timing, lazy performance-clock initialization, terminal descriptor coercion and redirected-stream behavior, emitted names, and TypeScript facades. Performance fields and BCL references remain scoped locals.
- Add 33 cases covering missing declarations and retry, independent availability, staged emission, saved IL, lazy clock state, descriptor coercion, emitter reuse, imports and aliases, CommonJS, async/generator use, and isolated command-line output.

Validation:

- Release test-project build: zero errors, seven existing NuGet vulnerability-feed warnings.
- All 449 focused tests passed, including all 33 new cases; no failures or skips. The eight isolated cases were rerun after making stdin redirection explicit for interactive runners.
- Code-quality gates: 28 existing duplicate groups, zero errors. Source and ownership audit passed, including assignment order, scoped fields, removed aliases, and file encoding preservation.
- Eleven baseline/current assemblies pass IL verification and preserve normalized metadata, native imports, literals, constants, and all 13,076 method bodies and exception regions. Nine execute with identical stdout/stderr; two validate hosted output.
- A separately captured pre-existing PerformanceObserver constructor IL error is unchanged. Existing observer regression tests also pass; this migration does not alter that facade behavior.

Continues #1599. Shared errors, workers/messages, language-runtime metadata, and the final residual ownership audit remain in the umbrella issue.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved reliability of performance timing and terminal checks when compiling and running programs.
  * Preserved clock state, timing behavior, and terminal descriptor handling across supported performance and TTY APIs.
  * Added validation to ensure these runtime features are fully initialized before use.

* **Compatibility**
  * Standalone compiled applications continue to run independently while supporting performance and terminal features.
  * Behavior remains consistent across TypeScript and CommonJS entry points, including edge-case inputs.

### CODE REVIEW

File: tests/SharpTS.Tests/CompilerTests/StandaloneDllTests.cs [hunk 1]
```diff
@@ -1319,6 +1319,98 @@ public void Isolated_ChildProcessMetadata_PreservesExecutionStreamsErrorsAndDepe
...
+    {
+        using var tempDir = IntegrationTests.CliTestHelper.CreateTempDirectory();
+        tempDir.CreateFile(entryPoint, source);
+        var dllPath = tempDir.GetPath("host_primitive_metadata.dll");
+        var compile = IntegrationTests.CliTestHelper.RunCli(
+            $"--no-tsconfig --compile \"{tempDir.GetPath(entryPoint)}\" -o \"{dllPath}\" --verify --standalone", tempDir.Path);
+        Assert.True(compile.ExitCode == 0, compile.StandardOutput + compile.StandardError);
+        Assert.DoesNotContain("SharpTS", GetAssemblyReferences(dllPath));
+        Assert.False(File.Exists(tempDir.GetPath("SharpTS.dll")));
+        Assert.Equal(expected, ExecuteCompiledDllIsolated(dllPath, timeoutMs: 15000,
```
[the remark this answers is not in the document]
Comment (nickna) [reply tests/SharpTS.Tests/CompilerTests/StandaloneDllTests.cs:1410]: Fixed in `b0fe6533`: the isolated host-primitive theory now passes `standardInput: ""`, which redirects stdin and closes it before execution. This makes the fd 0/null assertions independent of an interactive parent terminal. Release build and all eight affected isolated cases pass in a separate checkout. Runtime emission is unchanged.

**Comment by nickna** [comment]
@coderabbitai review
